File: backend/app/schemas/node.py
import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class NodeRegisterRequest(BaseModel):
    name: str


class NodeRegisterResponse(BaseModel):
    id: uuid.UUID
    name: str
    token: str


class NodeResponse(BaseModel):
    id: uuid.UUID
    name: str
    status: str
    last_heartbeat: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class NodeDeployRequest(BaseModel):
    package_version_id: uuid.UUID


class NodeDeploymentResponse(BaseModel):
    id: uuid.UUID
    node_id: uuid.UUID
    package_version_id: uuid.UUID
    status: str
    pending_params: dict[str, Any] | None
    last_result: dict[str, Any] | None
    last_run_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/node.ts
export interface EdgeNode {
  id: string
  name: string
  status: string
  last_heartbeat: string | null
  created_at: string
  updated_at: string
}

export interface NodeDeployment {
  id: string
  node_id: string
  package_version_id: string
  status: string
  pending_params: Record<string, unknown> | null
  last_result: Record<string, unknown> | null
  last_run_at: string | null
  created_at: string
}
